xarray/tests/__init__.py

```diff
@@ -356,12 +357,52 @@ def create_test_data(
 
 
 _CFTIME_CALENDARS = [
+    pytest.param(
+        cal, marks=pytest.mark.skipif(not has_cftime, reason="requires cftime")
+    )
+    for cal in sorted(
+        [
+            "365_day",
+            "360_day",
+            "julian",
+            "all_leap",
+            "366_day",
+            "gregorian",
+            "proleptic_gregorian",
+            "standard",
+        ]
+    )
+]
+
+_STANDARD_CALENDAR_NAMES = sorted(_STANDARD_CALENDARS_UNSORTED)
+_NON_STANDARD_CALENDAR_NAMES = {
+    "noleap",
     "365_day",
     "360_day",
     "julian",
     "all_leap",
     "366_day",
-    "gregorian",
-    "proleptic_gregorian",
-    "standard",
+}
+_NON_STANDARD_CALENDARS = [
+    pytest.param(
+        cal, marks=pytest.mark.skipif(not has_cftime, reason="requires cftime")
+    )
+    for cal in sorted(_NON_STANDARD_CALENDAR_NAMES)
 ]
+_STANDARD_CALENDARS = [pytest.param(cal) for cal in _STANDARD_CALENDAR_NAMES]
+_ALL_CALENDARS = sorted(_STANDARD_CALENDARS + _NON_STANDARD_CALENDARS)
+
```

Comment:
This might be a more concise way of defining `_CFTIME_CALENDARS`:
```suggestion
_CFTIME_CALENDARS = [pytest.param(*p.values, marks=requires_cftime) for p in _ALL_CALENDARS]
```